exasol_script_languages_container_tool/lib/api/language_activation.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+
+
+class SLCLanguage(Enum):
+    Java = "java"
+    Python3 = "python"
+    R = "r"
+
+
+@dataclass
+class SLCParameter:
+    key: str
+    value: str
+
+    def __str__(self) -> str:
+        return f"{self.key}={self.value}"
+
+
+@dataclass
+class LanguageDefinitionURL:
+    protocol: str
+    bucketfs_name: str
+    bucket_name: str
+    path_in_bucket: str
+    container_name: str
+    udf_client_path_within_container: str
+    parameters: List[SLCParameter]
+    language: SLCLanguage
+
+    def __str__(self) -> str:
+        parameters = ""
+        if self.parameters:
+            parameters = "&" + "&".join(str(p) for p in self.parameters)
+        path_in_bucket = self.path_in_bucket
+        if path_in_bucket and not path_in_bucket.endswith("/"):
+            path_in_bucket = f"{path_in_bucket}/"
+
+        return (
```

Comment:
Don't construct the url your self, please use the urlparse that also takes core about encoding. And regarding spaces they may actually work inside a definition of they are url encoded

Reply:
I can do that for the parameters. 

I also thought about using `urlpases` but for the whole URL I'am a bit afraid that changing the order of some things might break the `ALTER SESSION` command.

Reply:
Why should the order change.